File: src/jvmTest/java/org/hisp/dhis/lib/expression/StringExpressionTest.java
package org.hisp.dhis.lib.expression;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class StringExpressionTest {

    @Test
    void testEmpty() {
        assertEquals("", evaluate("''"));
    }

    @Test
    void testUnicode() {
        assertEquals("hello\n" +
                "world!\uD83D\uDE31 I think it works!", evaluate("'hello\\nworld!\\uD83D\\uDE31 I think it works!'"));
    }

    private static String evaluate(String expression) {
        return (String) new Expression(expression).evaluate();
    }
}

File: src/jvmTest/java/org/hisp/dhis/lib/expression/StringExpressionTest.kt
package org.hisp.dhis.lib.expression

import kotlin.test.assertEquals
import org.junit.jupiter.api.Test

internal class StringExpressionTest {
    @Test
    fun testEmpty() {
        assertEquals("", evaluate("''"))
    }

    @Test
    fun testUnicode() {
        assertEquals(
            """
    hello
    world!😱 I think it works!
    """.trimIndent(), evaluate("'hello\\nworld!\\uD83D\\uDE31 I think it works!'"))
    }

    companion object {
        private fun evaluate(expression: String): String? {
            return Expression(expression).evaluate() as String?
        }
    }
}
